source/non_matching/immersed_surface_quadrature.cc

```diff
@@ -0,0 +1,83 @@
+// ---------------------------------------------------------------------
+//
+// Copyright (C) 1998 - 2017 by the deal.II authors
+//
+// This file is part of the deal.II library.
+//
+// The deal.II library is free software; you can use it, redistribute
+// it, and/or modify it under the terms of the GNU Lesser General
+// Public License as published by the Free Software Foundation; either
+// version 2.1 of the License, or (at your option) any later version.
+// The full text of the license can be found in the file LICENSE at
+// the top level of the deal.II distribution.
+//
+// ---------------------------------------------------------------------
+
+#include <deal.II/non_matching/immersed_surface_quadrature.h>
+
+DEAL_II_NAMESPACE_OPEN
+namespace NonMatching
+{
+
+  template<int dim>
+  ImmersedSurfaceQuadrature<dim>::ImmersedSurfaceQuadrature(
+    const std::vector<Point<dim> > &points,
+    const std::vector<double> &weights,
+    const std::vector<Tensor <1,dim>> &normals):
+    Quadrature<dim>(points,weights),
+    normals(normals)
+  {
+    AssertDimension (weights.size(), points.size());
+    AssertDimension (normals.size(), points.size());
+#ifdef DEBUG
+    for (auto normal : normals)
+      {
+        Assert(std::abs(normal.norm() - 1.0) < 1e-9,
+               ExcMessage("Normal is not normalised"));
+      }
+#endif
```

Comment:
Minor comment: You can skip the `#ifdef DEBUG` here because the compiler will detect that the loop body is empty in release mode and hence optimize away the loop. Furthermore, I suggest you use 'normalized' rather than 'normalised' as we use American English spelling in the library.